target_chains/solana/programs/pyth-price-publisher/src/processor/initialize_publisher.rs

```diff
@@ -0,0 +1,267 @@
+use {
+    crate::{
+        accounts::{buffer, publisher_config},
+        ensure,
+        instruction::{InitializePublisherArgs, PUBLISHER_CONFIG_SEED},
+        validate::{
+            validate_authority, validate_buffer, validate_config, validate_publisher_config,
+            validate_system,
+        },
+    },
+    bytemuck::try_from_bytes,
+    solana_program::{
+        account_info::AccountInfo, entrypoint::ProgramResult, program::invoke_signed,
+        program_error::ProgramError, pubkey::Pubkey, rent::Rent, system_instruction,
+        sysvar::Sysvar,
+    },
+};
+
+// TODO: restrict access?
+pub fn initialize_publisher(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    data: &[u8],
+) -> ProgramResult {
+    let args: &InitializePublisherArgs =
+        try_from_bytes(data).map_err(|_| ProgramError::InvalidInstructionData)?;
+
+    let mut accounts = accounts.iter();
+    let first_account = accounts.next();
+    let config = validate_config(accounts.next(), args.config_bump, program_id, false)?;
+    let authority = validate_authority(first_account, &config)?;
+    let publisher_config = validate_publisher_config(
+        accounts.next(),
+        args.publisher_config_bump,
+        &args.publisher.into(),
+        program_id,
+        true,
+    )?;
+    let buffer = validate_buffer(accounts.next(), program_id)?;
+    let system = validate_system(accounts.next())?;
+
+    // Deposit enough tokens to allocate the account.
+    let rent = Rent::get()?;
+    let lamports = rent.minimum_balance(publisher_config::SIZE);
+
+    invoke_signed(
+        &system_instruction::create_account(
+            authority.key,
+            publisher_config.key,
+            lamports,
+            publisher_config::SIZE
+                .try_into()
+                .expect("unexpected overflow"),
+            program_id,
+        ),
+        &[authority.clone(), publisher_config.clone(), system.clone()],
+        &[&[
+            PUBLISHER_CONFIG_SEED.as_bytes(),
+            &args.publisher,
+            &[args.publisher_config_bump],
+        ]],
+    )?;
+
+    let mut publisher_config_data = publisher_config.data.borrow_mut();
+    publisher_config::create(
+        *publisher_config_data,
+        args.publisher,
+        buffer.key.to_bytes(),
+    )?;
+
+    // Write an initial Header into the buffer account to prepare it to receive prices.
```

Comment:
Do we want to have any minimum size check for the buffer?

Reply:
Right now we just check if the size is enough to fit the header. The code should work fine with any size.